template.Debian/docker-entrypoint.sh

```diff
@@ -0,0 +1,238 @@
+#!/bin/bash
+set -eo pipefail
+shopt -s nullglob
+
+# if command starts with an option, prepend mysqld
+if [ "${1:0:1}" = '-' ]; then
+	set -- mysqld "$@"
+fi
+
+# skip setup if they want an option that stops mysqld
+wantHelp=
+for arg; do
+	case "$arg" in
+		-'?'|--help|--print-defaults|-V|--version)
+			wantHelp=1
+			break
+			;;
+	esac
+done
+
+# usage: file_env VAR [DEFAULT]
+#    ie: file_env 'XYZ_DB_PASSWORD' 'example'
+# (will allow for "$XYZ_DB_PASSWORD_FILE" to fill in the value of
+#  "$XYZ_DB_PASSWORD" from a file, especially for Docker's secrets feature)
+file_env() {
+	local var="$1"
+	local fileVar="${var}_FILE"
+	local def="${2:-}"
+	if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
+		echo >&2 "$(date --rfc-3339=seconds) [ERROR] [Entrypoint]: Both $var and $fileVar are set (but are exclusive)"
```

Comment:
This all seems to use `echo [>&2] "$(date --rfc-3339=seconds) [ERROR/Note] [Entrypoint]: ...` quite a lot; do you think it'd make sense to make some simple helpers out of that so we stay consistent?  Something like:

```bash
_log() {
	echo "$(date --rfc-3339=seconds) [$1] [Entrypoint]: $*"
}
_note() {
	_log Note "$@"
}
_error() {
	_log ERROR "$@" >&2
}
```

Reply:
Yes, definitely. Will add it :)

Reply:
Also added a _warn (used it for the notice about files in the initdb.d directory being ignored)